frontend-v2/src/components/nav.tsx

```diff
@@ -0,0 +1,148 @@
+'use client'
+
+import navbarData from '../../../frontend/nav.json'
+import { fetchSession } from '@/app/session'
+import { User } from 'lucide-react'
+import { useCallback, useState } from 'react'
+import Image from 'next/image'
+import logo1 from '../../../frontend/static/img/logo1.png'
+import { cn } from '@/lib/utils'
+import Link from 'next/link'
+
+export const Navbar = ({
+  pageName,
+  session,
+}: {
+  /** The name of the active page, corresponding to the name in Vue. */
+  pageName: string
+  /** The user sesion returned from the `fetchSession` function. */
+  session: Awaited<ReturnType<typeof fetchSession>>
+}) => {
+  const [isMobileExpanded, setMobileExpanded] = useState(false)
+  const [activeDropdown, setActiveDropdown] = useState<string | null>(null)
+
+  const hasAccess = useCallback(
+    (roleRequired: string[] | undefined) => {
+      if (!roleRequired) return true
+
+      const role = session?.user?.role
+      if (!role) return false
+
+      return roleRequired.some((it) => {
+        if (it === 'admin') return role === 'admin'
+        if (it === 'organizer') return role === 'admin' || role === 'organizer'
+        if (it === 'attendance')
+          return (
+            role === 'admin' || role === 'organizer' || role === 'attendance'
+          )
+        if (it === 'non-sfbay') return role === 'non-sfbay'
+        return false
+      })
+    },
```

Comment:
could this be extracted as a named function, perhaps in the top level of the module, since it isn't super navigation or React related? perhaps called getHasAccess.

Reply:
sure i can extract it

Reply:
you can write the unit tests :P